libmscore/rendermidi.cpp

```diff
@@ -1527,74 +1527,28 @@ void Score::createPlayEvents()
-            else {
-                  numerator   /= 3;
-                  clickTicks  *= 3;
-                  }
-            }
+      qreal tempo = tempomap()->tempo(msrTick);
+      TimeSigFrac timeSig = sigmap()->timesig(msrTick).nominal();
 
-      // NUMBER OF TICKS
-      int numOfClicks = numerator;                          // default to a full measure of 'clicks'
-      int lastPause   = clickTicks;                         // the number of ticks to wait after the last 'click'
-      int pickupClickOffset = 0;
-
-      if (countIn) {
-            // if not at the beginning of a measure, add clicks for the initial measure part
-            if (msrTick < playPos) {
-                  int delta    = playPos - msrTick;
-                  int addClick = (delta + clickTicks - 1) / clickTicks;     // round num. of clicks up
-                  numOfClicks += addClick;
-                  lastPause    = delta - (addClick - 1) * clickTicks;       // anything after last click time is final pause
-                  }
-            // or if measure not complete (anacrusis), add clicks for the missing measure part
-            else if (m->ticks() < clickTicks * numerator) {
-                  int delta    = clickTicks * numerator - m->ticks();
-                  int addClick = (delta + clickTicks - 1) / clickTicks;
-                  numOfClicks += addClick;
-                  lastPause    = delta - (addClick - 1) * clickTicks;
-                  }
-            }
-      else {
-            // compute pickup offset to avoid wrong tick sound
-            if (sig.nominal() != sig.timesig()) {
-                  numOfClicks = msrTicks / clickTicks;
-                  pickupClickOffset = numerator - numOfClicks;
-                  }
-            }
 
-      // click-clack-clack triplets
-      if (triplets)
-            numerator = 3;
-      int tick = 0;
-      NPlayEvent event;
-      for (int i = 0; i < numOfClicks; i++) {
-            tick = (countIn ? 0 : m->tick()) + i * clickTicks + tickOffset;
-            // do not add a tick if the clickticks is greater than the measure duration
-            // except in countIn
-            event.setType(((i + pickupClickOffset) % numerator) == 0 ? ME_TICK1 : ME_TICK2);
-            events->insert(std::pair<int,NPlayEvent>(tick, event));
+      int clickTicks = timeSig.isBeatedCompound(tempo) ? timeSig.beatTicks() : timeSig.dUnitTicks();
+      int endTick = m->endTick();
+
+      int rtick;
+
+      if (m->isAnacrusis()) {
+            int rem = m->ticks() % clickTicks;
+            msrTick += rem;
+            rtick = rem + timeSig.ticksPerMeasure() - m->ticks();
             }
-      return tick + lastPause;
+      else
+            rtick = 0;
+
+      for (int tick = msrTick; tick < endTick; tick += clickTicks, rtick+=clickTicks)
+            events->insert(std::pair<int,NPlayEvent>(tick + tickOffset, MScore::seq->beatType2metronomeEvent(timeSig.rtick2beatType(rtick))));
```

Comment:
Hmm, this only seems to break when running the test. It works fine for playback (even with the metronome enabled) and no complaints during MIDI export. An advantage of having the function in mscore is that the tick and tack sounds could potentially be made adjustable in the preferences.